### PR TITLE
Fix tests and other inconsistencies.

### PR DESCRIPTION
**Pull request by squaresurf**
Why:

* A new project from this template should pass tests and use the latest
software.

This change addresses the need by:

* Fix .editorconfig max line length to be the same as the rubocop.yml.
* Update Rails.
* Update slim_lint and rubocop since it's a dependency.
* Add default slim lint config so that it's easier to change line length
for those that would like to do so.
* Add missing database_cleaner gem (used by the factory girl linter).
* Fix slim lint errors in linted files.
* Fix the admin_users factory as it was missing required fields.

Side effects:

* Tests should pass in a new repo.

### CODE REVIEW

**Comment by squaresurf** [comment]
I would appreciate an in depth review of the slim template changes as I might have done something wrong fixing the linter errors.

**Comment by bbrock25** [comment]
:+1: what was breaking the tests?

**Comment by squaresurf** [comment]
@bbrock25 the following broke tests:

* Missing `database_cleaner` gem.
* Old version of slim_lint that tied to a rubocop version that expected the rails cops to be defined differently in the `.rubocop.yml`.
* Slim lint violations.
* The incomplete factory.

**Comment by bbrock25** [comment]
Thanks!

File: app/views/admin/_edit_buttons.html.slim [hunk 1]
```diff
@@ -1,11 +1,21 @@
 ol.edit-buttons
   li
-    = link_to "Banner", atomic_cms.edit_component_path("banner"), class: "button"
+    = link_to("Banner",
+      atomic_cms.edit_component_path("banner"),
+      class: "button")
```
Comment (samcdavid) [inline app/views/admin/_edit_buttons.html.slim]: Did slim lint alter these?

**Comment by samcdavid** [comment]
@squaresurf, just the one question :smile:

File: app/views/admin/_edit_buttons.html.slim (hunk 1 above)
Comment (squaresurf) [inline app/views/admin/_edit_buttons.html.slim]: No, I did it by hand. Slim lint doesn't have an autofix that I know of.

File: Rakefile [hunk 2]
```diff
@@ -1,12 +1,15 @@
 require File.expand_path("../config/application", __FILE__)
 require "slim_lint/rake_task"
+require 'rubocop/rake_task'
```
Comment (Mrjaco12) [inline Rakefile]: Single quotes here.

File: app/views/components/banner.html.slim [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+          class=add_option(options[:background_image], "has-bg-image")
+        ]
           - if options[:icon]
             .icon
-              img src="#{options[:icon]}" alt="#{options[:title]}"
-          h2.banner-title[class="#{options[:text_size]}"] #{options[:title]}
-          .body[class="#{options[:text_size]}"] #{markdown(options[:content])}
+              img src=options[:icon] alt=options[:title]
+          h2.banner-title[class=options[:text_size]] #{options[:title]}
+          .body[class=options[:text_size]] #{markdown(options[:content])}
```
Comment (Mrjaco12) [inline app/views/components/banner.html.slim]: Did we test this? I'm not sure the vars are getting interpolated.

Comment (squaresurf) [inline app/views/components/banner.html.slim]: I wasn't sure how to fully test that.

File: Rakefile (hunk 2 above)
Comment (squaresurf) [inline Rakefile]: Good catch.

**Comment by Mrjaco12** [comment]
I also think we should upate slim_lint/rubocop as they're a little outdated.

**Comment by squaresurf** [comment]
@Mrjaco12 I did update slim_lint/rubocop :smile:

**Comment by Mrjaco12** [comment]
Ahh. I see. Good job.

**Comment by Mrjaco12** [comment]
:+1:

**Comment by squaresurf** [comment]
@Mrjaco12 should I figure out how to test all the slim templates? As far as I understand it, they should all still work as expected.

**Comment by Mrjaco12** [comment]
I checked it out and it looks like it's working for me.

**Comment by samcdavid** [comment]
Out of curiosity, I checked master and the latest tag `v0.3.4` and master is ahead of the latest tag. Should  a new tag be created or is this still WIP?

**Comment by squaresurf** [comment]
I agree that a new tag should be created. I will do so now.

**Comment by squaresurf** [comment]
I've created and pushed a tag `v0.4.0`. I incremented the number a minor version as we added features to the bin scripts.

**Comment by samcdavid** [comment]
:+1:

**Pull request merged**